.github/workflows/cleanup-azure-resources.yml

```diff
@@ -1,37 +1,123 @@
+name: CleanupAzureResources
+
 on:
   pull_request:
     types: [closed]
+  #push:
+  #  branches: [users/v-shivakumar/AB#86411]
 
-name: CleanupAzureResources
+permissions:
+   id-token: write
+   contents: write
 
 jobs:
   cleanup:
-    runs-on: ubuntu-latest
+    runs-on: windows-latest # AzureAD PS module requires Windows.
+
     steps:
+      - name: Checkout Current Branch to access Custom Powershell Modules
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          #ref: ${{ github.ref }}
+          ref: 'main' # we are only interested in importing the custom PS modules from main branch
+
       - name: Login via Az module
         uses: azure/login@v1.4.1
         with:
-          creds: ${{ secrets.AZURE_CREDENTIALS }}
+          client-id: "${{secrets.AZ_DEVOPS_APP_ID}}"
+          tenant-id: "${{secrets.AZ_DEVOPS_TENANT_ID}}"
+          subscription-id: "${{secrets.AZ_DEVOPS_SUBSCRIPTION_ID}}"
```

Comment:
We cant view GitHub secrets. How will someone know what app Id it is? Do you still think we should keep these ids as secret

Reply:
Documented in Wiki and referred the wiki link in the yml comment.